src/libcollections/str.rs

```diff
@@ -923,11 +923,12 @@ pub trait StrExt: Index<RangeFull, Output = str> {
 
     /// Returns a string with all pre- and suffixes that match a pattern repeatedly removed.
     ///
-    /// The pattern can be a simple `&str`, or a closure that determines the split.
+    /// The pattern is required to be a `DoubleEndedSearcher`, therefore `&str` patterns cannot
+    /// be used.
```

Comment:
This lost the 'closure' aspect. I would change this to read:

> The pattern can be any `DoubleEndedSearcher`, including a closure that determines the split.

This keeps it overall more consistent with the rest of the other examples.